### ISSUE TITLE
No error on missing protocol generic argument

### ISSUE THREAD

**Issue by Voldemat** [astral-sh/ty#4215]
### Summary

this code does not produce any errors from ty:
```python
from contextlib import AbstractAsyncContextManager
from types import TracebackType
from typing import Self


class Check(AbstractAsyncContextManager):
    async def __aenter__(self) -> Self:
        return self

    async def __aexit__(
        self,
        exc_type: type[BaseException] | None,
        exc_value: BaseException | None,
        traceback: TracebackType | None,
    ) -> None:
        pass
```
while mypy shows this error:
```
check.py:6: error: Missing type arguments for generic type "AbstractAsyncContextManager"  [type-arg]
    class Check(AbstractAsyncContextManager):
                ^
Found 1 error in 1 file (checked 1 source file)
```

Adding type argument solves the issue for mypy:
```python
from contextlib import AbstractAsyncContextManager
from types import TracebackType
from typing import Self


class Check(AbstractAsyncContextManager["Check"]):
    async def __aenter__(self) -> Self:
        return self

    async def __aexit__(
        self,
        exc_type: type[BaseException] | None,
        exc_value: BaseException | None,
        traceback: TracebackType | None,
    ) -> None:
        pass
```

### Version

ty 0.0.69 (1dca8443e 2026-08-06)

**Comment by carljm** [comment]
ty supports this diagnostic, but it is opt-in (disabled by default). It is disabled by default in mypy also. If you are seeing this diagnostic in mypy, that means you have a mypy configuration which enables `disallow-any-generics` or `strict` options.

For ty the needed configuration is

```toml
[tool.ty.rules]
missing-type-argument = "error"
```

You might be interested in https://docs.astral.sh/ty/coming-from-mypy-or-pyright/#stricter-checking-with-ty
